src/cli.rs

```diff
@@ -213,6 +254,15 @@ fn parse_year_range(value: &str) -> Result<Vec<YearRange>, &'static str> {
         .collect::<Result<Vec<YearRange>, &'static str>>()
 }
 
+// A simple function to manage similaries across multiple cases.
+// May ultimately be generalised to a function to manage all progress UX
+// that can be switched on and off.
+fn print_metrics_count(search_results: SearchResults) -> usize {
+    let len_requests = search_results.0.shape().0;
+    println!("Found {len_requests} metric(s).");
```

Comment:
Potentially this, but could be included in the next PR for other refinements:
```suggestion
    if len_requests == 1 {
        println!("Found {len_requests} metric.");
    } else {
        println!("Found {len_requests} metrics.");
    }
```